moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -33,74 +33,407 @@
+      servo_params.publish_joint_velocities)
+  {
+    RCLCPP_ERROR(LOGGER, "When publishing a std_msgs/Float64MultiArray, "
+                         "you must select positions OR velocities."
+                         "Check the parameters YAML file used to launch this node.");
+    params_valid = false;
+  }
+
+  if (servo_params.scene_collision_proximity_threshold < servo_params.self_collision_proximity_threshold)
+  {
+    RCLCPP_ERROR(LOGGER, "Parameter 'self_collision_proximity_threshold' should probably be less "
+                         "than or equal to 'scene_collision_proximity_threshold'."
+                         "Check the parameters YAML file used to launch this node.");
+    params_valid = false;
+  }
+
+  return params_valid;
+}
 
-  // Crunch the numbers in this timer
-  servo_calcs_.start();
+bool Servo::updateParams()
+{
+  bool params_updated = false;
+
+  if (servo_param_listener_->is_old(servo_params_))
+  {
+    auto params = servo_param_listener_->get_params();
+
+    const bool params_valid = validateParams(params);
+    if (params_valid)
+    {
+      if (params.override_velocity_scaling_factor != servo_params_.override_velocity_scaling_factor)
+      {
+        RCLCPP_INFO_STREAM(LOGGER, "override_velocity_scaling_factor changed to : "
+                                       << std::to_string(params.override_velocity_scaling_factor));
+      }
+      if (params.move_group_name != servo_params_.move_group_name)
+      {
+        RCLCPP_INFO_STREAM(LOGGER, "Move group changed from " << servo_params_.move_group_name << " to "
+                                                              << params.move_group_name);
+      }
 
-  // Check collisions in this timer
-  if (servo_params_.check_collisions)
-    collision_checker_.start();
+      servo_params_ = params;
+      params_updated = true;
+    }
+    else
+    {
+      RCLCPP_WARN_STREAM(LOGGER, "Parameters will not be updated.");
+    }
+  }
+  return params_updated;
+}
+
+servo::Params& Servo::getParams()
+{
+  return servo_params_;
 }
 
-void Servo::stop()
+StatusCode Servo::getStatus()
 {
-  servo_calcs_.stop();
-  collision_checker_.stop();
+  return servo_status_;
 }
 
-bool Servo::getCommandFrameTransform(Eigen::Isometry3d& transform)
+const std::string Servo::getStatusMessage()
 {
-  return servo_calcs_.getCommandFrameTransform(transform);
+  return SERVO_STATUS_CODE_MAP.at(servo_status_);
 }
 
-bool Servo::getCommandFrameTransform(geometry_msgs::msg::TransformStamped& transform)
+CommandType Servo::expectedCommandType()
 {
-  return servo_calcs_.getCommandFrameTransform(transform);
+  return expected_command_type_;
 }
 
-bool Servo::getEEFrameTransform(Eigen::Isometry3d& transform)
+void Servo::expectedCommandType(const CommandType& command_type)
 {
-  return servo_calcs_.getEEFrameTransform(transform);
+  expected_command_type_ = command_type;
 }
```

Comment:
It's confusing that these both have the same name... maybe call them get/set?

Reply:
Done